domain/secretbackend/service/service_test.go

```diff
@@ -146,6 +145,12 @@ func (s *serviceSuite) setupMocks(c *gc.C) *gomock.Controller {
 	s.mockSepicalSecretProvider = NewMockSecretsBackend(ctrl)
 	s.mockStringWatcher = NewMockStringsWatcher(ctrl)
 
+	mockAtomicContext := NewMockAtomicContext(ctrl)
+	mockAtomicContext.EXPECT().Context().Return(context.Background()).AnyTimes()
+	s.mockState.EXPECT().RunAtomic(gomock.Any(), gomock.Any()).DoAndReturn(func(_ context.Context, f func(domain.AtomicContext) error) error {
+		return f(mockAtomicContext)
```

Comment:
`return fn(domaintesting.NewAtomicContext(ctx))`

Then we can remove the mock atomic context above. But we won't need this now if GetSecretBackendDetails is not atomic